Expose member names of serializable types to the IPC testing API

gavin-apple opened the pull request:
#### fee2513ebe872d50402d5d1cc9a6190865efe46a
<pre>
Expose member names of serializable types to the IPC testing API
<a href="https://bugs.webkit.org/show_bug.cgi?id=250678">https://bugs.webkit.org/show_bug.cgi?id=250678</a>
rdar://104297785

Reviewed by Alex Christensen.

Expose member names of individual types to the IPC testing API in order
to provide context to callers about the values of the fields to provide.

* Source/WebKit/Scripts/generate-serializers.py:
(generate_serialized_type_info):
* Source/WebKit/Shared/SerializedTypeInfo.h:
* Source/WebKit/WebProcess/WebPage/IPCTestingAPI.cpp:
(WebKit::IPCTestingAPI::JSIPC::serializedTypeInfo):

Canonical link: <a href="https://commits.webkit.org/259112@main">https://commits.webkit.org/259112@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/81266f3f5348b82c2e55af92e924bcbb2b26494e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/103603 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/12718 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/36556 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/112836 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/173174 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/13744 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/3617 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/95854 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/112002 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/109375 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/10589 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/93651 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/38322 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/107099 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/92413 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/25254 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/79980 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/6094 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/26655 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/6275 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/3182 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/12256 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/46166 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6274 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/8027 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

saambarati commented on Source/WebKit/Shared/SerializedTypeInfo.h:
```diff
@@ -32,9 +32,14 @@
+};
+
 struct SerializedTypeInfo {
     ASCIILiteral name;
```
How is memberName different than name?

gavin-apple replied:
If we declare:

```
class X {
    int a;
};
```

SerializedTypeInfo::name == "X"
SerializedType::memberType == "int"
SerializedType::memberName == "a"

gavin-apple committed:
Expose member names of serializable types to the IPC testing API
https://bugs.webkit.org/show_bug.cgi?id=250678
rdar://104297785

Reviewed by Alex Christensen.

Expose member names of individual types to the IPC testing API in order
to provide context to callers about the values of the fields to provide.

* Source/WebKit/Scripts/generate-serializers.py:
(generate_serialized_type_info):
* Source/WebKit/Shared/SerializedTypeInfo.h:
* Source/WebKit/WebProcess/WebPage/IPCTestingAPI.cpp:
(WebKit::IPCTestingAPI::JSIPC::serializedTypeInfo):

Canonical link: https://commits.webkit.org/259112@main
